genMOSC_Torture declaration causes multiple build issues

There appears to be two problems with the lines:
https://github.com/CTSRD-CHERI/QuickCheckVEngine/blob/9fc7ec5c2790f8074dab6c77a0b2cf28f2543ced/src/QuickCheckVEngine/Templates/GenTransExec.hs#L174-L175

Firstly, the function names do not match each other. This results in the following error:
```
src/QuickCheckVEngine/Templates/GenTransExec.hs:175:1: error:
    Multiple declarations of ‘genBSC_Excps_Torture’
    Declared at: src/QuickCheckVEngine/Templates/GenTransExec.hs:148:1
                 src/QuickCheckVEngine/Templates/GenTransExec.hs:175:1
    |
175 | genBSC_Excps_Torture arch tmpReg = random $ do
    | ^^^^^^^^^^^^^^^^^^^^
Error: cabal: Failed to build exe:QCVEngine from QCVEngine-0.1.0.0.
```
Secondly, when the names are corrected, the following type error occurs:
```
src/QuickCheckVEngine/Templates/GenTransExec.hs:174:20: error:
    Not in scope: type constructor or class ‘ArchDesc’
    |
174 | genMOSC_Torture :: ArchDesc -> Integer -> Template
    |                    ^^^^^^^^
Error: cabal: Failed to build exe:QCVEngine from QCVEngine-0.1.0.0.
```

--- a/src/QuickCheckVEngine/Templates/GenTransExec.hs
+++ b/src/QuickCheckVEngine/Templates/GenTransExec.hs
@@ -51,6 +51,7 @@ import RISCV.RV32_Xcheri
 import RISCV.RV32_F
 import RISCV.RV32_D
 import RISCV.RV_CSRs
+import RISCV.ArchDesc
 import QuickCheckVEngine.Template
 import QuickCheckVEngine.Templates.GenMemory
 import QuickCheckVEngine.Templates.Utils
@@ -172,7 +173,7 @@ genBSC_Excps_Torture tmpReg = random $ do
 
 
 genMOSC_Torture :: ArchDesc -> Integer -> Template
-genBSC_Excps_Torture arch tmpReg = random $ do
+genMOSC_Torture arch tmpReg = random $ do
   imm <- bits 12
   longImm <- bits 20
   src1 <- sbcRegs

